app/utils/print/sections/AnswerRowConverter.scala

```diff
@@ -39,10 +39,29 @@ class AnswerRowConverter @Inject()(checkAnswersFormatters: CheckAnswersFormatter
     question(query, userAnswers, labelKey, format, messageArg)
   }
 
-  def utr(userAnswers: UserAnswers,
-          labelKey: String,
-          messageArg: String = "")
-         (implicit messages: Messages): Option[AnswerRow] = {
+  def whichIdentifier(userAnswers: UserAnswers)
+                     (implicit messages: Messages): Option[AnswerRow] = {
+
+    userAnswers.identifierType match {
+      case UTR =>
+        Some(AnswerRow(
+          messages(s"whichIdentifier.checkYourAnswersLabel"),
+          HtmlFormat.escape(messages("uniqueTaxReference.checkYourAnswersLabel")),
+          None
+        ))
+      case URN =>
+        Some(AnswerRow(
+          messages(s"whichIdentifier.checkYourAnswersLabel"),
+          HtmlFormat.escape(messages("uniqueReferenceNumber.checkYourAnswersLabel")),
+          None
+        ))
+    }
```

Comment:
Could this be refactored to
```
val prefix = if (userAnswers.identifierType == UTR) "uniqueTaxReference" else "uniqueReferenceNumber"
Some(AnswerRow(
  messages(s"whichIdentifier.checkYourAnswersLabel"),
  HtmlFormat.escape(messages(s"$prefix.checkYourAnswersLabel")),
  None
))
```
or similar?

Reply:
Yes, yes it could!